common/utils.c

```diff
@@ -836,6 +838,19 @@ TDNFJoinPath(char **ppszPath, ...)
          * before stripping all leading slashes */
         if (i == 0)
         {
+            /* we also check to see if the base path is a remote URL; for URLs
```

Comment:
I don't really like that you are using `TDNFJoinPath()` for this, which is a generic function to join path elements (much like python's `os.path.join()`. I think it's not very efficient, because this code will now be used in all places where we join paths, which are mostly local.

Instead, I think we should create another version of this function that is only used where we need this functionality  of preserving the URL query string.

Reply:
@adithyaj , are you goin g to address my comments?